### PR TITLE
Use hub-ci locally

### PR DESCRIPTION
**Pull request by lhoestq**
I switched the HF endpoint to the hub-ci one and added the corresponding tokens, and fixed the docker-compose to use the right base.
Now the local dev environment from `make dev-start` works correctly.

Close https://github.com/huggingface/datasets-server/issues/765

### CODE REVIEW

File: workers/datasets_based/src/datasets_based/worker_loop.py [hunk 1]
```diff
@@ -97,6 +98,8 @@ def sleep(self) -> None:
         time.sleep(duration)
 
     def loop(self) -> None:
+        if os.environ.get("HF_ENDPOINT"):
```
Comment (severo) [inline workers/datasets_based/src/datasets_based/worker_loop.py:101]: We don't access the os environment variables outside of the config.py files. Do you think it's possible to keep on with this practice and move the log there?

**Review by severo — APPROVED** [review]
OK

I will let you define what to do with the comment about accessing the env vars.

File: workers/datasets_based/src/datasets_based/worker_loop.py (hunk 1 above)
Comment (lhoestq) [reply workers/datasets_based/src/datasets_based/worker_loop.py]: I kept the logging here and got the endpoint value from worker_factory.app_config.common

Comment (severo) [reply workers/datasets_based/src/datasets_based/worker_loop.py]: Perfect, thanks!

**Pull request merged**